api/policies/hasGameId.js

```diff
@@ -0,0 +1,19 @@
+/**
+ * hasGameId
+ *
+ * @module      :: Policy
+ * @description :: Only allows requests that contain a 'gameId' parameter, which is a number
+ * @docs        :: http://sailsjs.org/#!/documentation/concepts/Policies
+ *
+ */
+module.exports = function (req, res, next) {
+  // User is allowed, proceed to the next policy,
+  // or if this is the last policy, the controller
+  if (Object.hasOwnProperty.call(req.body, 'gameId')) {
+    if (typeof req.body.gameId === 'number') return next();
+  }
```

Comment:
The original policies were written before optional chaining was available, but now that we've updated to node < 14, we can consolidate this to:
```suggestion
  if (typeof req?.body?.gameId === 'number') return next();
```